### PR TITLE
Add manticore adapter

### PR DESCRIPTION
**Pull request by mikeknep**
Adds an adapter for [Manticore](https://github.com/cheald/manticore), addressing #454.

I don't know if there is any functionality unique to Manticore that warrants additional tests beyond the shared "with WebMock" suite. @cheald, any thoughts?

Manticore specs pass locally on JRuby 1.7.8 and JRuby 9.0.0.0. Running the full suite on both JRuby versions yields 36 failing http_rb_adapter specs (this is also the case on master branch).

I noticed that each adapter's spec_helper is handling array values in headers in unique but similar ways. I think a helper could be added to `WebMock::Util::Headers` to address this duplication. However, rather than add that and refactor across libraries now, I figured it'd be better to keep this PR focused exclusively on Manticore. (I bring this up because this PR adds duplicate `join_array_values` methods in the adapter and spec_helper.)

### CODE REVIEW

File: lib/webmock/http_lib_adapters/manticore_adapter.rb [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+          end
+
+          def join_array_values(headers)
+            headers.reduce({}) do |h, (k,v)|
+              v = v.join(', ') if v.is_a?(Array)
+              h.merge(k => v)
+            end
+          end
+
+          def response_object_for(client, request, context, &block)
```
Comment (cheald) [inline lib/webmock/http_lib_adapters/manticore_adapter.rb]: This is a private method that doesn't seem to be called anywhere; is this correct?

Comment (cheald) [inline lib/webmock/http_lib_adapters/manticore_adapter.rb]: *sigh* nevermind, it's overriding a Manticore method. I should go make :coffee:

**Comment by cheald** [comment]
To be honest I'm probably the wrong person to evaluate the feature coverage; I haven't used WebMock since the early Rails 2 days :)

Manticore does expose an awful lot of functionality, though, but it tries to have a relatively sane and compact (read: testable) interface. I'm happy to advise on specific examples if the WebMock maintainers can enumerate some potentially questionable cases.

**Comment by mikeknep** [comment]
Update: added support for stubbing requests made via Manticore's facade methods (i.e. calling `Manticore.get(url)` instead of `client = Manticore::Client.new; client.get(url)`).

@bblimke any thoughts on this getting merged in?

**Comment by bblimke** [comment]
@mikeknep I'm won't be able to take care of this until 12th September. @presidentJFK perhaps you fancy looking after it?

**Comment by presidentJFK** [comment]
I don't have any experience using Manticore, so my help might be a little slow. But if no one gets this sooner, I'll take a look once I finish some other Webmock todos.

**Comment by mikeknep** [comment]
Checking in on this. @bblimke or @presidentJFK is there anything you need from me or that I can help with for this PR?

**Comment by presidentJFK** [comment]
@mikeknep this is next up on my list and I will looking at this week, probably starting tonight. 

So I will be hitting you up for questions/feedback soon. Thanks!

**Comment by bblimke** [comment]
Thanks. Everything looks good!

It's only mandatory for each adapter to pass "with WebMock" examples. Additional specs are only needed if given http client provides additional functionalities not covered by "with WebMock" specs.

@mikeknep I hope you can help with Manticore adapter maintenance in case Manticore changes? :)

**Comment by mikeknep** [comment]
Awesome, glad to hear it! And I'll certainly help check out any problems with it in the future.

**Comment by presidentJFK** [comment]
@mikeknep
I open a new Pull Request #523 the fix the CI Test Suite.

**Comment by presidentJFK** [comment]
closing this now that #523 is merged in. Thanks @mikeknep!

**Pull request closed without merging**